tests/config_test.go

```diff
@@ -201,21 +201,22 @@ func (c *config) Check(
 func TestInferCustomCheckConfig(t *testing.T) {
 	t.Parallel()
 
-	s, err := integration.NewServer(t.Context(),
-		"test",
-		semver.MustParse("0.0.0"),
-		integration.WithProvider(infer.Provider(infer.Options{
-			Config: infer.Config[*config](),
-		})),
-	)
-	require.NoError(t, err)
-
```

Comment:
Worth calling out these tests introduced a race condition because we're now invoking methods on the user-provided config struct, and the tests are calling Configure in parallel. IMO it would be reasonable to error if the provider gets a second Configure call, but I'd like to handle that separately.